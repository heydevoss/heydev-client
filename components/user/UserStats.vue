<template>
  <div class="content">
    <div v-for="stat in stats" :key="stat.label" class="stat">
      <span class="label">
        {{ stat.label }}
      </span>
      <span class="count">
        {{ stat.count }}
      </span>
    </div>
  </div>
</template>
<script>
export default {
  name: 'UserStats',
  components: {},
  props: {
    totalPullRequests: Number,
    totalCommits: Number,
    totalIssues: Number,
    firstContributionDate: String
  },
  data() {
    return {
      stats: [
        { label: 'Pull Requests', count: this.totalPullRequests },
        { label: 'Issues', count: this.totalCommits },
        { label: 'Commits', count: this.totalIssues },
        { label: 'First Contribuition', count: this.firstContributionDate }
      ]
    }
  }
}
</script>
<style lang="scss" scoped>
.content {
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
}

.count {
  color: orange;
  text-align: center;
  font-size: 1.5rem;
}
.stat {
  display: flex;
  flex-direction: column;
  margin: 10px 10px;
}
</style>
