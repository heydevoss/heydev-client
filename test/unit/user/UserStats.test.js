import { shallowMount } from '@vue/test-utils'
import UserStats from '@/components/user/UserStats'

describe('<UserStats/> Component', () => {
  const wrapper = shallowMount(UserStats, {
    propsData: {
      totalPullRequests: 5,
      totalCommits: 34,
      totalIssues: 21,
      firstContributionDate: '2018-04-04'
    }
  })

  it('should be a Vue instance and renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the stats labels', () => {
    expect(wrapper.html()).toContain('Pull Requests')
    expect(wrapper.html()).toContain('Issues')
    expect(wrapper.html()).toContain('Commits')
  })

  it('should render the stats values', () => {
    expect(wrapper.html()).toContain('5')
    expect(wrapper.html()).toContain('34')
    expect(wrapper.html()).toContain('21')
    expect(wrapper.html()).toContain('2018-04-04')
  })

  it('should render without errors when no prop is passed', () => {
    wrapper.setProps({
      totalPullRequests: null,
      totalCommits: null,
      totalIssues: null,
      firstContributionDate: null
    })
    expect(wrapper.isEmpty()).toBeFalsy()
  })
})
