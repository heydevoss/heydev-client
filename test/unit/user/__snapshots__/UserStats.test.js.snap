// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStats/> Component should be a Vue instance and renders correctly 1`] = `
<div
  class="content"
>
  <div
    class="stat"
  >
    <span
      class="label"
    >
      
      Pull Requests
    
    </span>
     
    <span
      class="count"
    >
      
      5
    
    </span>
  </div>
  <div
    class="stat"
  >
    <span
      class="label"
    >
      
      Issues
    
    </span>
     
    <span
      class="count"
    >
      
      21
    
    </span>
  </div>
  <div
    class="stat"
  >
    <span
      class="label"
    >
      
      Commits
    
    </span>
     
    <span
      class="count"
    >
      
      34
    
    </span>
  </div>
  <div
    class="stat"
  >
    <span
      class="label"
    >
      
      First Contribuition
    
    </span>
     
    <span
      class="count"
    >
      
      2018-04-04
    
    </span>
  </div>
</div>
`;
